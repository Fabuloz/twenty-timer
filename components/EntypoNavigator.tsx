import React, { memo, ReactElement } from 'react';
import { TouchableOpacity, View } from 'react-native';
import { Entypo } from '@expo/vector-icons';
import { scale } from 'react-native-size-matters';
import AppStyle from '../styles/AppStyles';

// EntypoNavigator component prop types
interface EntypoNavigatorProps {
  name: string;
  onPress: () => boolean;
}

/**
 * Navigation icon
 * @param name - The name of the icon to render.
 */
const EntypoNavigator = ({
  name,
  onPress
}: EntypoNavigatorProps): ReactElement => {
  return (
    <TouchableOpacity onPress={onPress} style={AppStyle.navigator}>
      <View>
        <Entypo name={name} size={scale(50)} style={AppStyle.icon} />
      </View>
    </TouchableOpacity>
  );
};

export default memo(EntypoNavigator);
