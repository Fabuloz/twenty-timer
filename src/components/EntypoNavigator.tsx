import React, { memo, ReactElement } from 'react';
import { TouchableOpacity, View, TouchableOpacityProps } from 'react-native';
import { Entypo } from '@expo/vector-icons';
import { scale } from 'react-native-size-matters';
import AppStyles from '../styles/AppStyles';

// EntypoNavigator component prop types
interface EntypoNavigatorProps extends TouchableOpacityProps {
  name: string;
}

/**
 * Navigation icon
 * @param name - The name of the icon to render.
 */
const EntypoNavigator = ({
  name,
  ...rest
}: EntypoNavigatorProps): ReactElement => {
  return (
    <TouchableOpacity
      testID="entypo-navigator"
      style={AppStyles.navigator}
      {...rest}
    >
      <View>
        <Entypo name={name} size={scale(50)} style={AppStyles.icon} />
      </View>
    </TouchableOpacity>
  );
};

export default memo(EntypoNavigator);
