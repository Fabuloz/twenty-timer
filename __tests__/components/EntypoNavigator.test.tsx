import React from 'react';
import { fireEvent, render, toJSON } from '@testing-library/react-native';
import EntypoNavigator from 'components/EntypoNavigator';

test('renders and matches snapshot', (): void => {
  const { container, getByTestId } = render(
    <EntypoNavigator name="chevron-up" />
  );

  expect(getByTestId('entypo-navigator')).toBeTruthy();
  expect(toJSON(container)).toMatchSnapshot();
});

test('spreads props', (): void => {
  const props = {
    accessibilityLabel: 'entypo-navigator',
    name: 'chevron-up',
    onPress: jest.fn((): void => {})
  };
  const { getByLabelText } = render(<EntypoNavigator {...props} />);

  // Sets accessibilityLabel
  const entypoNavigator = getByLabelText(props.accessibilityLabel);
  expect(entypoNavigator).toBeTruthy();

  // Sets onPress method
  fireEvent.press(entypoNavigator);
  expect(props.onPress).toHaveBeenCalled();
});
