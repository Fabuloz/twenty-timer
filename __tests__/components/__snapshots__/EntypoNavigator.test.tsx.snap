// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and matches snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    style={
      Object {
        "bottom": "5%",
        "position": "absolute",
        "right": "5%",
      }
    }
    testID="entypo-navigator"
  >
    <View>
      <Text />
    </View>
  </TouchableOpacity>
</View>
`;
